aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,264 @@
+        :param message_type: the expected message type
+        :param kwargs: other expected message attributes
+
+        :return: boolean result of the evaluation and accompanied message
+        """
+        if (
+            type(actual_message) != message_type  # pylint: disable=unidiomatic-typecheck
+        ):
+            return (
+                False,
+                "The message types do not match. Actual type: {}. Expected type: {}".format(
+                    type(actual_message), message_type
+                ),
+            )
+
+        for attribute_name, expected_value in kwargs.items():
+            attribute = getattr(actual_message, attribute_name)
+            if callable(attribute):
+                if attribute != expected_value:
+                    return (
+                        False,
+                        "The '{}' fields do not match. Actual '{}': {}. Expected '{}': {}".format(   # pylint: disable=duplicate-string-formatting-argument
+                            attribute_name,
+                            attribute_name,
+                            attribute,
+                            attribute_name,
+                            expected_value,
+                        ),
+                    )
+
+        return True, "The message has the provided expected attributes."
+
+    def build_incoming_message(
+        self,
+        message_type: Type[Message],
+        performative: Message.Performative,  # pylint: disable=unused-argument
+        dialogue_reference: Optional[  # pylint: disable=unused-argument
+            Tuple[str, str]
+        ] = None,
+        message_id: Optional[int] = None,  # pylint: disable=unused-argument
+        target: Optional[int] = None,  # pylint: disable=unused-argument
+        to: Optional[Address] = None,
+        sender: Address = "counterparty",
+        **kwargs,
+    ) -> Message:
+        """
+        Quickly create an incoming message with the provided attributes.
+
+        For any attribute not provided, the corresponding default value in message is used.
+
+        :param message_type: the type of the message
+        :param dialogue_reference: the dialogue_reference
+        :param message_id: the message_id
+        :param target: the target
+        :param performative: the performative
+        :param to: the 'to' address
+        :param sender: the 'sender' address
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        message_attributes = dict()
+        message_attributes["dialogue_reference"] = Dialogues._generate_dialogue_nonce()  # pylint: disable=protected-access
+        for arg_name in inspect.getfullargspec(self.build_incoming_message)[0][1:]:
+            if (
+                arg_name not in {"message_type", "to", "sender"}
+                and locals()[arg_name] is not None
+            ):
+                message_attributes[arg_name] = locals()[arg_name]
```

Comment:
I don't like the usage of locals...can lead to hard to debug errors if the method is extended. Also, pylint above shows you're not using the keyword arguments..just use them properly and avoid using locals.

Reply:
Done